camviewer_ui_impl.py

```diff
@@ -1297,6 +1307,44 @@ def onSizeUpdate(self):
         except Exception:
             pass
 
+    def user_set_max_image_rate(self, rate: int) -> None:
+        """
+        Call set_max_image_rate and record the value as the desired rate.
+
+        This desired rate is used so that after a timeout, we can
+        restore the GUI to the user's last desired rate when they
+        interact with the GUI again.
+        """
+        self.last_des_max_rate = rate
+        self.set_max_image_rate(rate)
+
+    def set_max_image_rate(self, rate: int) -> None:
+        """
+        Update the maximum allowed rate by restarting the appropriate timer.
+
+        Rate is expected to be a positive integer in Hz.
+
+        When this timer expires, wantImage is called.
+        If a new image is available at this time, it will be fetched and rendered.
+
+        The image rate will determine the rate limiting timeout.
+        1 Hz will never time out
+        5 Hz or less will time out in a week
+        30 Hz and greater will time out in a day
+        """
+        rate = int(rate)
+        if rate <= 0:
+            raise ValueError("Rate must be greater than zero!")
+        self.imageTimer.start(int(1000.0 / rate))
+        self.ui.label_max_rate_value.setText(f"{rate} Hz")
+
+        if rate <= 1:
+            self.stop_disco()
+        else:
+            self.setDisco(
+                np.interp(rate, [5, 30], [self.max_timeout, self.min_timeout])
```

Comment:
So between 5 and 30, we interpolate between the max and min timeouts we have set, and outside of that we are capped at the max or min timeouts.  

A nifty little shortcut 👍